### PR TITLE
Get specific user

### PR DESCRIPTION
**Pull request by ZohaibManzoor00**
Created a utility function to check if inputs are empty in utils.ts - will be useful for future

### CODE REVIEW

File: src/services/users.service.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { isEmpty } from '@/utils/util';
+
+class UserService {
+  public async getUsersFromDB(): Promise<BaseUserDto[]> {
+    const users: BaseUserDto[] = await UserEntity.find();
+    return users;
+  }
+
+  public async getSpecificUserFromDB(ID: number): Promise<BaseUserDto> {
+    // if (isEmpty(ID)) throw new HttpException(400, 'USER ID IS EMPTY');
```
Comment (raylu511) [inline src/services/users.service.ts:13]: what's this doing here if you're commenting it out?

File: src/services/users.service.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { HttpException } from '@/exceptions';
+import { isEmpty } from '@/utils/util';
+
+class UserService {
+  public async getUsersFromDB(): Promise<BaseUserDto[]> {
+    const users: BaseUserDto[] = await UserEntity.find();
+    return users;
+  }
+
+  public async getSpecificUserFromDB(ID: number): Promise<BaseUserDto> {
```
Comment (raylu511) [inline src/services/users.service.ts:12]: lowercase id?

File: src/routes/users.route.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  public router = Router();
+  public usersController = new UsersController();
+
+  constructor() {
+    this.initializeRoutes();
+  }
+
+  private initializeRoutes() {
+    this.router.get(`${this.path}`, this.usersController.getUsers);
+    this.router.get(`${this.path}/:id`, this.usersController.getSpecificUser)
```
Comment (raylu511) [inline src/routes/users.route.ts:17]: should unauthenticated users be able to query our endpoints?

File: src/controllers/users.controller.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      const userData: BaseUserDto[] = await this.usersService.getUsersFromDB();
+      res.status(200).json({ data: userData });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public getSpecificUser = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const ID = Number(req.params.id);
```
Comment (raylu511) [inline src/controllers/users.controller.ts:19]: we don't usually capitalize variable names unless it's for stuff like 
(e.g. BASE_URL, DB_NAME,etc.) at the top file level.
I would keep this as lowercase id.

File: src/controllers/users.controller.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      const userData: BaseUserDto = await this.usersService.getSpecificUserFromDB(ID);
```
Comment (raylu511) [inline src/controllers/users.controller.ts:20]: nit: we can name this `data `and we can just do `res.status(200).json({ data });`

File: src/controllers/users.controller.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      const data: BaseUserDto[] = await this.usersService.getUsersFromDB();
+      res.status(200).json({ data });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public getSpecificUser = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const user_id = Number(req.params.id);
```
Comment (raylu511) [inline src/controllers/users.controller.ts:19]: we shouldn't mix casing here.

File: src/app.ts [hunk 7]
```diff
@@ -64,11 +64,11 @@ class App {
...
-      await mongoDataSource.initialize();
-    } catch (error) {
-      console.error(error);
-    }
+    // try {
+    //   await mongoDataSource.initialize();
+    // } catch (error) {
+    //   console.error(error);
+    // }
   }
```
Comment (DLiSWE) [inline src/app.ts:72]: uncomment?

**Review by Matt-Tan15 — APPROVED** [review]

File: src/services/users.service.ts [hunk 8]
```diff
@@ -1,11 +1,20 @@
...
 
 class UserService {
   public async getUsersFromDB(): Promise<BaseUserDto[]> {
     const users: BaseUserDto[] = await UserEntity.find();
     return users;
   }
+
+  public async getSpecificUserFromDB(userId: number): Promise<BaseUserDto> {
+    const user: BaseUserDto = await UserEntity.findOne({ where: { id: userId } });
+    if (!user) throw new HttpException(409, `USER WITH ID ${userId} DOES NOT EXIST`);
```
Comment (DLiSWE) [inline src/services/users.service.ts:13]: nit: all caps on the http exception is pretty different from what the others are. Maybe we should discuss how exceptions look.

**Review by DLiSWE — APPROVED** [review]

**Review by emir4ik-code — DISMISSED** [review]

**Review by raylu511 — APPROVED** [review]

**Review by Matt-Tan15 — APPROVED** [review]

**Pull request merged**